### PR TITLE
Don't let the actual status code bleed through.

### PR DESCRIPTION
**Pull request by dereuromark**
Code with jQuery success() as http://sandbox3.dereuromark.de/sandbox/ajax-examples/redirecting used to work.
The changes in the beforeRedirect() AJAX catching broke that as the status code bleed-through makes jQuery not workable anymore. It jumps to error() callback instead - and the response data is not accessible anymore.

### CODE REVIEW

**Comment by markstory** [comment]
Seems reasonable, and the kind of thing that might warrant a test so we don't accidentally lose it again.

File: src/Controller/Component/RequestHandlerComponent.php [hunk 1]
```diff
@@ -300,6 +300,7 @@ public function beforeRedirect(Event $event, $url, Response $response)
                 'REQUEST_METHOD' => 'GET'
             ]
         ]));
+        $response->statusCode(200);
         $response->send();
         $response->stop();
     }
```
Comment (lorenzo) [inline src/Controller/Component/RequestHandlerComponent.php]: This part of cake is still a horrible idea in my mind :S

Comment (dereuromark) [inline src/Controller/Component/RequestHandlerComponent.php]: Unless one returns with a full JSON object :) That contains all the data at once.

Comment (lorenzo) [inline src/Controller/Component/RequestHandlerComponent.php]: let's do that!

Comment (lorenzo) [inline src/Controller/Component/RequestHandlerComponent.php]: By "Let's do that" I meant, return the response object with the right code instad of `send()` and `stop()`

Comment (dereuromark) [inline src/Controller/Component/RequestHandlerComponent.php]: Oh no. That would make my [AJAX Component](https://github.com/dereuromark/cakephp-ajax/blob/master/src/Controller/Component/AjaxComponent.php#L100) obsolete :trollface: ^^

Comment (dereuromark) [inline src/Controller/Component/RequestHandlerComponent.php]: Ah, now I understand, you mean returning the object and let the controller resolve the sending?

Comment (lorenzo) [inline src/Controller/Component/RequestHandlerComponent.php]: yes

Comment (dereuromark) [inline src/Controller/Component/RequestHandlerComponent.php]: Wouldn't that be a different PR, as this PR and its change has to be done either way to reset the status code back to 200?

Comment (lorenzo) [inline src/Controller/Component/RequestHandlerComponent.php]: Why not both at the same time? the changes are quite minimal

Comment (dereuromark) [inline src/Controller/Component/RequestHandlerComponent.php]: It breaks BC though, does it not? Returning an object now and handling it from the controller? I wonder if we should merge this and apply the proposed refactoring for 3.1 instead. I would be willing to do that.

Comment (lorenzo) [inline src/Controller/Component/RequestHandlerComponent.php]: It does not have to be handled in the controller, returning the response here will bubble up to the dispatcher

**Comment by dereuromark** [comment]
Refactored. Tests pass.

**Comment by markstory** [comment]
Nice work :+1:
